src/lib.rs

```diff
@@ -158,6 +158,7 @@ pub async fn init(mode: SystemMode) -> Result<(), Error> {
 
     let mut buffer = [0; 64];
     mode.create_at_command(&mut buffer)?;
+    mode.setup_psm().await?;
     at::send_at_bytes::<0>(&buffer).await?;
 
     Ok(())
```

Comment:
It's a bit weird to put this here right in between the creation of a buffer, filling it and then using it. Please move this to after the block

```suggestion
    let mut buffer = [0; 64];
    mode.create_at_command(&mut buffer)?;
    at::send_at_bytes::<0>(&buffer).await?;

    mode.setup_psm().await?;

    Ok(())
```